### PR TITLE
 Milestone 9: Models 

### PR DESCRIPTION
**Pull request by AbigiyaTY**
## Work Portfolio  Milestone 9: Models

In this milestone I:

- Add div in the HTML file to hold the Javascript
- created HTML file using Javascript
- style the Javascript using CSS

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -177,11 +176,19 @@ <h2>Multi-Post Stories</h2>
...
+          See Project
+        </button>
         </div>
+
+        <div id="displayWork" class="displayWork"></div>
+        <div id="overlay"></div>
+
         <div class="work-portofolio-2">
             <div class="grid-container grid-container2">
                 <div class="grid-item1 ml"></div>
```
Comment (tresorsawasawa) [inline index.html]: - [ ] Kindly make sure you remove all the hard code for the projects list like in this  **[example](https://github.com/olawale-o/olawale-o.me/pull/15/files#diff-0eb547304658805aad788d320f10bf1f292797b5e6d745a3bf617584da017051R95)** `i.e.: from LINE 159 - 278(index.html file)` then go in your `JavaScript` file and create an array of objects to store JavaScript objects containing the information for each project.

File: js/script.js [hunk 2]
```diff
@@ -2,20 +2,95 @@ const mainMenu = document.querySelector('.mainMenu');
...
+<img class="displayImg large" src="${desktopModalContent.displayImgLarge}" alt="Project Image on desktop device">
+<img class="displayImg small" src="${mobileModalContent.displayImgSmall}" alt="Project Image on mobile device">
+<p class="displayP">${desktopModalContent.descriptionDesktop}</p>
+<ul class="diplayUl">${langArray}</ul>
+<ul class="diplayUlLG">${langArrayLg}</ul>
+<div class="footerDis">
+<button class="buttonDis buttonOne" type="button"><a href="https://abigiyaty.github.io/Portofolio-/" target="_blank">See Live <i class="disFA fa fa-chevron-right"></i></a></button>
+<button class="buttonDis buttonTwo" type="button"><a href="https://github.com/AbigiyaTY/Portofolio-" target="_blank">See Source <i class="disFA fa fa-github"></i></a></button>
+</div>`;
+
```
Comment (tresorsawasawa) [inline js/script.js]: - [ ] Kindly make sure you implement **an array of objects** to store JavaScript objects containing the information for each project. And each object contains at least the following information about the project: `name`, `description`, `featured image`, `technologies`,` link to live version`, and `link to the project source code`. i.e.:
```javascript
const projects = [
 {
    name: "Submarine game",
    description: "A Phaser3 platform game built as part of the Microverse JavaScript program.",
    image: "link_to_image.jpg",
    technologies: ["Phaser3", "JavaScript", "Jest"],
    source_link: "link_to_live_version.com",
    live_link: "link_to_live_version.com",
},
...
]
```

- [ ] Also make sure the projects section is created dynamically with the information from the projects array (all the HTML of the section is created when the page loads). **i.e. As I mentioned in the previous comment HTML code for the projects list should be removed, and instead it should be added via JavaScript.**

Comment (tresorsawasawa) [inline js/script.js]: - [ ] Once you made the above change, kindly make sure Modal popup data is populated with the data from the projects array and not assigned value manually.

- [ ] Also make sure when the user clicks (or taps) the **See Project** button to check project details, the popup with details about the project appears. Right now the  **See Project** button displays an empty **blue background**
Here is the result once clicked:
Screenshot from 2022-07-22 02-18-15

Comment (tresorsawasawa) [inline js/script.js]: - [ ] Kindly make sure when **Modal popup** appears and the user clicks (or taps) the close (X) button, the popup disappears.

- [ ] Kindly make sure the popup is responsive, **it fits mobile and desktop sizes properly** and follows the design for both desktop and mobile versions from [the chosen template in Figma](https://www.figma.com/file/l7SqJ3ZfkAKih9sFxvWSR4/Microverse-Student-Project-1?node-id=0%3A1) (e.g., font, colors, images, text, margins,...) like on the images below(mobile and desktop):

| Popup Mobile Figma Design | Popup Desktop Figma Design |
| ----------------------------- | ------------------------------- |
| Screenshot from 2022-07-22 02-27-47 | Screenshot from 2022-07-22 02-29-18 |

**Review by tresorsawasawa — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES :recycle::recycle: 

Hi @
Good job so far! I know you worked hard to implement this project, I really appreciate the effort you put in! But there are some issues that you still need to work on to go to the next project but you are almost there! 

almost-there

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Your design looks good :100: :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Follow GitHub flow :heavy_check_mark: 

## Required changes :recycle: :recycle: 
- [ ] **Kindly make sure you are using **[HTML-CSS-JS](https://github.com/microverseinc/linters-config/tree/master/html-css-js)** linters as required for the project and not **HTML-CSS**. You can check the here **[HTML-CSS-JS](https://github.com/microverseinc/linters-config/tree/master/html-css-js)** to make their confugartion.**
- [ ] Check the comments under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: js/script.js (hunk 2 above)
Comment (muneebulrehman) [reply js/script.js]: Kindly make sure to implement the changes requested by the previous reviewer. You have not done so yet. Your whole work section should be generated by javascript. Consider removing the hardcoded data from index.html and generate this data dynamically. If you have any difficulty with understanding the requirements of the project, feel free to leave a message on slack, I go by the same name as on GitHub. 😊

File: js/script.js [hunk 3]
```diff
@@ -2,20 +2,95 @@ const mainMenu = document.querySelector('.mainMenu');
...
+  });
+});
+
+function closeModal(modal) {
+  if (modal == null) return;
+  modal.classList.remove('active');
+  overlay.classList.remove('active');
+}
+
+overlay.addEventListener('click', () => {
```
Comment (muneebulrehman) [inline js/script.js]: When closing the modal, it shows this error. Consider fixing it.
image

Comment (muneebulrehman) [reply js/script.js]: Also when I close the modal, it takes me back to the top of the page. Kindly fix this issue.

**Review by muneebulrehman — CHANGES_REQUESTED** [review]
Hi @AbigiyaTY , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- Descriptive PR

 ## Required Changes ♻️
_Check the comments under the review._

- [ ] You have not added the javascript linter check. Kindly do so and then ask for a re-review. Javascript linter check is a must for any project containing javascript code.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: index.html [hunk 4]
```diff
@@ -177,11 +178,17 @@ <h2>Multi-Post Stories</h2>
...
-            <button class="button1" type="button">See Project</button>
+            <button data-modal-target="#displayWork" class="button1" type="button">
+          See Project
+        </button>
         </div>
+        <div id="displayWork" class="displayWork"></div>
+        <div id="overlay"></div>
         <div class="work-portofolio-2">
             <div class="grid-container grid-container2">
                 <div class="grid-item1 ml"></div>
```
Comment (HAMMAS-SALEEM) [inline index.html:194]: - [ ] You have a hardcode in your index.html file for all the projects you are showing on your portfolio. Please remove this code and generate projects from javascript dynamically :blush: as the previous reviewers requested :+1:
- [ ] When you close your modal popup then it refreshes the page. This is not the expected behavior. This should only close the popup not refresh the page as the previous reviewers requested :thinking:

File: index.html [hunk 5]
```diff
@@ -202,10 +209,14 @@ <h2>Multi-Post Stories</h2>
...
+            </button>
                 </div>
             </div>
-            <button class="button1" type="button">See Project</button>
+            <button data-modal-target="#displayWork" class="button1" type="button">
+          See Project
+        </button>
         </div>
         <div class="work-portofolio-3">
             <div class="grid-container grid-container1 grid-container3">
```
Comment (HAMMAS-SALEEM) [inline index.html:222]: ### Your design
popup-overflow-auto-issue

- [ ] You can see in your design that your popup contents are going down of your popup window and they are not accessible I mean your popup is not scrollable. Please make it scrollable by adding **overrflow: auto;**

File: js/popup.js [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+displayWork.innerHTML = `<div class="headerDis"><h2 class="displayHeader">${desktopModalContent.title}</h2><span> <a onclick="closePopupModal()" href="" id='fixed' class="closeWork">${desktopModalContent.closeWork}</a></span></div>
+<img class="displayImg large" src="${desktopModalContent.displayImgLarge}" alt="Project Image on desktop device">
+<img class="displayImg small" src="${mobileModalContent.displayImgSmall}" alt="Project Image on mobile device">
+<p class="displayP">${desktopModalContent.descriptionDesktop}</p>
+<ul class="diplayUl">${langArray}</ul>
+<ul class="diplayUlLG">${langArrayLg}</ul>
+<div class="footerDis">
+<button class="buttonDis buttonOne" type="button"><a href="https://abigiyaty.github.io/Portofolio-/" target="_blank">See Live <i class="disFA fa fa-chevron-right"></i></a></button>
+<button class="buttonDis buttonTwo" type="button"><a href="https://github.com/AbigiyaTY/Portofolio-" target="_blank">See Source <i class="disFA fa fa-github"></i></a></button>
+</div>`;
```
Comment (HAMMAS-SALEEM) [inline js/popup.js]: - [ ] You are making the popup window :blush: but you are not doing it right. You should show data on a popup through an array of objects which will contain all your projects. :+1:

File: js/popup.js [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+  diplayUl: ['|', 'html', '|', 'Ruby on rails', '|', 'css'],
+}];
+
+const desktopModalContent = [{
+  title: 'Multi-post stories',
+  closeWork: '<i class="closeWork fa fa-times"></i>',
+  displayImgLarge: './images/Snapshoot-Portfolio-lg.png',
+  descriptionDesktop: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it 1960s with the releaLorem Ipsum is simply dummy text of the printing and typesetting  ever since the 1500s, when an unknown printer took a galley of type veris lapoa todoe.',
+  diplayUl: ['|', 'html', '|', 'Ruby on rails', '|', 'css', '|', 'Github'],
+}];
```
Comment (HAMMAS-SALEEM) [inline js/popup.js]: - [ ] You are making an array of objects but you should not make different objects for mobile and desktop. Please make only one object for both orientations :+1:
- [ ] Add all the projects here not just one :+1:

**Review by HAMMAS-SALEEM — CHANGES_REQUESTED** [review]
# Required Changes :recycle:
almost-there

### Hey  @AbigiyaTY :wave:

### Highlights :1st_place_medal:
 
:heavy_check_mark: Following HTML, CSS best practices
:heavy_check_mark: Popup is responsive
:heavy_check_mark: Descriptive PR title
:heavy_check_mark: Descriptive PR summary
:heavy_check_mark: Descriptive Branch name

Good job on implementing some of the changes requested by the previous reviewer, there are some issues that you still need to work on to go to the next project but you are almost there! :blush:

# Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @HAMMAS-SALEEM in your question so I can receive the notification.

_**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_

------------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: index.html (hunk 4 above)
Comment (AbigiyaTY) [reply index.html:194]: - Hellow Hammas thanks for the review, the first button is for the mobile view and the second button is for the desktop view I didn't use the hard button to generate the javascript popup it has its own script kindly can you check the popup.js file line 41 and 42 thank you.

File: index.html (hunk 5 above)
Comment (AbigiyaTY) [reply index.html:222]: - Thanks, it works.

File: js/popup.js (hunk 6 above)
Comment (AbigiyaTY) [reply js/popup.js]: - Thanks for the feedback, I have created an array for both the Mobile version(popup.js 4 - 10) and the desktop version (popup.js 12 - 18) if I am wrong can you explain more, please.

File: js/popup.js (hunk 7 above)
Comment (AbigiyaTY) [reply js/popup.js:18]: - Thanks for your feedback I can create one array for both but the value or pair for the list and image is different so how can I fix that? popup.js(line 7 and 15, lines 9 and 17) thanks so much.

File: js/popup.js (hunk 6 above)
Comment (HAMMAS-SALEEM) [reply js/popup.js]: Ok come on slack, I can explain everything in a single call if you want? :blush:

**Review by KayLemba — APPROVED** [review]
# STATUS: APPROVED :green_circle: :clap: 

Hello  @AbigiyaTY      , 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

HIGHLIGHTS:

- Your project is professional 
- Your repo is professional 
- Well structured files 


Now you can merge this branch and move on :1st_place_medal: 
HAPPY CODING AND GOOD LUCK WITH THE COMMING ONES :crossed_fingers: :four_leaf_clover: 

well-1


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**